src/Stratis.Bitcoin.Features.Consensus/Checkpoints.cs

```diff
@@ -182,8 +187,16 @@ public bool CheckHardened(int height, uint256 hash)
         public CheckpointInfo GetCheckpoint(int height)
         {
             CheckpointInfo checkpoint;
-            this.checkpoints.TryGetValue(height, out checkpoint);
+            this.GetCheckpoints().TryGetValue(height, out checkpoint);
             return checkpoint;
         }
+
+        private Dictionary<int, CheckpointInfo> GetCheckpoints()
+        {
+            if (this.ConsensusSettings == null || !this.ConsensusSettings.UseCheckpoints)
```

Comment:
Can the `ConsensusSettings ` be null here?

Reply:
Yes it can if you call the parameterless constructor.